dataframe-geo/src/main/kotlin/org/jetbrains/kotlinx/dataframe/geo/geocode/Geocoder.kt

```diff
@@ -0,0 +1,121 @@
+package org.jetbrains.kotlinx.dataframe.geo.geocode
+
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.cio.CIO
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.client.statement.bodyAsText
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+import io.ktor.serialization.kotlinx.json.json
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import org.jetbrains.kotlinx.dataframe.api.dataFrameOf
+import org.jetbrains.kotlinx.dataframe.geo.GeoDataFrame
+import org.jetbrains.kotlinx.dataframe.geo.toGeo
+import org.locationtech.jts.geom.Geometry
+import org.locationtech.jts.geom.GeometryFactory
+import org.locationtech.jts.io.geojson.GeoJsonReader
+
+
+object Geocoder {
+
+    private val url = "https://geo2.datalore.jetbrains.com/map_data/geocoding"
+
+    private fun countryQuery(country: String) = """ {
+    "region_query_names" : [ "$country" ],
+    "region_query_countries" : null,
+    "region_query_states" : null,
+    "region_query_counties" : null,
+    "ambiguity_resolver" : {
+      "ambiguity_resolver_ignoring_strategy" : null,
+      "ambiguity_resolver_box" : null,
+      "ambiguity_resolver_closest_coord" : null
+    }
+    }
+    """.trimIndent()
+
+    private fun geocodeQuery(countries: List<String>) = """
```

Comment:
This is exactly where the linter will help :)

Reply:
Done